SOD.Common/Helpers/InputDetection.cs

```diff
@@ -1,19 +1,526 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using BepInEx;
+using Rewired;
+using SOD.Common.Custom;
+using SOD.Common.Extensions;
+using UnityEngine;
+using UniverseLib.Utility;
 
 namespace SOD.Common.Helpers
 {
     public sealed class InputDetection
     {
+        private const int KEYBOARD_AND_MOUSE_CONTROLLER_ID = 0;
+        private readonly List<string> CONTROLLER_MAP_CATEGORIES = ["Interaction", "Movement", "Menu", "CityEdit"];
+
         internal InputDetection() { }
 
+        public Rewired.Player RewiredPlayer => InputController.Instance.player;
+
         /// <summary>
-        /// Raised when a button's state changes from up/released to down/pressed and vice versa.
+        /// Get the action name associated with an InteractionKey in a format that is useable with Rewired GetButton methods.
+        /// </summary>
+        /// <param name="interactionKey"></param>
+        /// <returns></returns>
+        public string GetRewiredActionName(InteractablePreset.InteractionKey interactionKey)
+        {
+            var gameMappedKey = Enum.GetName(typeof(InteractablePreset.InteractionKey), interactionKey);
+            var capitalizedKey = $"{gameMappedKey.Substring(0, 1).ToUpper()}{gameMappedKey.Substring(1)}";
+            return capitalizedKey;
+        }
+
+        /// <summary>
+        /// Get the Rewired InputAction associated with an InteractionKey.
+        /// </summary>
+        /// <param name="interactionKey"></param>
+        /// <returns></returns>
+        public InputAction GetRewiredAction(InteractablePreset.InteractionKey interactionKey)
+        {
+            return ReInput.MappingHelper.Instance.GetAction(GetRewiredActionName(interactionKey));
+        }
+
+        /// <summary>
+        /// Get the Rewired controller binding currently bound to an InteractionKey's Rewired InputAction.
+        /// </summary>
+        /// <param name="interactionKey"></param>
+        /// <returns></returns>
+        public ActionElementMap GetBinding(InteractablePreset.InteractionKey interactionKey)
+        {
+            List<ActionElementMap> controllerMappings = KeyboardAndMouseMappings;
+            try
+            {
+                return controllerMappings.First(x => x.actionId == GetRewiredAction(interactionKey).id);
+            }
+            catch (InvalidOperationException)
+            {
+                Plugin.Log.LogDebug($"Could not find action controller binding for interactionKey: {interactionKey}");
+                return null;
+            }
+            catch (NullReferenceException)
+            {
+                Plugin.Log.LogDebug($"Null reference exception for interactionKey: {interactionKey}");
```

Comment:
Not a big fan of this try catches, can we just handle the errors properly with readable messages, null reference doesn't say much for a user of the lib